src/lib/reasoners/shostak.ml

```diff
@@ -51,6 +67,7 @@ struct
 
   type rview =
```

Comment:
This is somewhat true. I wrote it, and would like to keep it, this way for two reasons:
 - It avoids nested recursion (type-level recursion inside a module-level recursion; the `rview` type already depends recursively on `CX.r` through the different Shostak modules)
 - Uniform with the use at the term level where we call e.g. `CX.print`; otherwise we would have to make all functions in this module recursive themselves.

Reply:
Indeed, let us keep your code so.